lib/srv/discovery/discovery.go

```diff
@@ -115,6 +117,13 @@ type gcpInstaller interface {
 type Config struct {
 	// CloudClients is an interface for retrieving cloud clients.
 	CloudClients cloud.Clients
+
+	// FetchersClients gets the AWS clients for the given region for the fetchers.
+	FetchersClients fetchers.ClientGetter
```

Comment:
IF we reserve FetchersClients for AWS, it should be more clear about it.
We should probably also split the cloud clients as well

Reply:
This is the idea indeed, EC2 and SSM have been split out from cloud clients already, this extracts out completely EKS and partially STS. I'll rename to be more explicit about AWS.

Reply:
Done.